org-roam.el

```diff
@@ -1407,6 +1407,31 @@ included as a candidate."
   (interactive)
   (find-file (seq-random-elt (org-roam--list-all-files))))
 
+(defun org-roam-insert--delete-region (region &optional unshield reinsert)
+  "Delete REGION in `org-roam-insert' caller.
+
+When UNSHIELD is non-nil, removes the read-only property.
+
+When REINSERT is non-nil, reinsert the original text."
+  (when region
+    (pcase-let ((`(,min . ,max) region))
+      (when unshield
+        (let ((inhibit-read-only t))
+          (remove-text-properties min max '(read-only t))))
+      (delete-region min max)
+      (when reinsert
+        (save-excursion
+          (goto-char min)
+          (insert (org-roam-capture--get :link-description))))
```

Comment:
Does reinsert change the text? why not just not delete?

Reply:
It's to handle `org-not-abort` cases.  Originally, I only removed the `[[` and `]]`, but, it made it harder to modularise the code.  So, since we were already storing `:link-description`, and the region cannot be edited whilst the `org-roam-capture` process is active,  I figured it was fine to do it that way.  I could remove the arg and use `(when (org-note-abort) …) instead.

Reply:
> I could remove the arg and use (when (org-note-abort) …) instead.

Not too sure about this, actually.  I thought it'd hurt modularity, but since we're only using it `org-roam-insert`…

Not a fan of this function.